test/e2e/smoke-test/packages/contour/1.18.2/contour-test.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+# Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+
+# Checking package is installed or not
+tanzu package installed list | grep "contour.community.tanzu.vmware.com"
+exitcode_contour=$?
+MY_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+echo "${MY_DIR}"
+if [ "${exitcode_contour}" == 1 ] 
+then
+    version=$(tanzu package available list contour.community.tanzu.vmware.com | tail -n 1 | awk '{print $2}')
+    tanzu package install contour --package-name contour.community.tanzu.vmware.com -f "${MY_DIR}"/contour-values.yaml --version "${version}"
+fi
+sleep 15s
+
+# Providing prerequisite 
+kubectl create namespace contour-example-workload
+kubectl create deployment nginx-example --image nginx --namespace contour-example-workload
+kubectl create service clusterip nginx-example --tcp 80:80 --namespace contour-example-workload
+
+kubectl apply -f - <<EOF
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: nginx-example-proxy
+  namespace: contour-example-workload
+  labels:
+    app: ingress
+spec:
+  virtualhost:
+    fqdn: nginx-example.projectcontour.io
+  routes:
+  - conditions:
+    - prefix: /
+    services:
+    - name: nginx-example
+      port: 80
+EOF
+
+sleep 15s
+kubectl --namespace projectcontour port-forward svc/envoy 5436:80 &
```

Comment:
Ideally I would expect to close this port-forward process once the test is done. Currently it's left hanging. In CI pipeline this is fine maybe - as CI environment is throw away environment which is killed off once the pipeline is done, but if someone runs this in local, and runs many times, it can leave many processes running behind the scenes unnecessarily

Reply:
One idea is to capture the process ID of the process that's running the background and then kill it later once the test is done, like this

```bash
kubectl --namespace projectcontour port-forward svc/envoy 5436:80 &
pid=$1
trap '{ kill -9 $pid; }' EXIT
```

Reply:
This is an improvement that we can do in this PR or separate PR, just putting it out here

Reply:
I think there is a automatic disconnection after few seconds but will cross check this. Thanks @karuppiah7890

Reply:
From what I know, disconnections might happen with "broken pipe" errors, but the process generally keeps running. `kubectl port-forward` is an interactive command and stops only when one does `ctrl+c` so if it runs in background, one has to use a process manager to kill the process

Reply:
Ok ok I will resolve this too.

Reply:
Looks like the process does die at some point, like this -

```bash
Handling connection for 5436
E1103 08:46:10.247676   79180 portforward.go:400] an error occurred forwarding 5436 -> 8080: error forwarding port 8080 to pod 1ad3d14fe335373c62096e65be6307d739c59e06e2575d0a0fa8e90595742166, uid : failed to find sandbox "1ad3d14fe335373c62096e65be6307d739c59e06e2575d0a0fa8e90595742166" in store: does not exist
E1103 08:47:14.360122   79180 portforward.go:233] lost connection to pod

[1]+  Done                    kubectl --namespace projectcontour port-forward svc/envoy 5436:80
```

I think this happens when all the resources are deleted - the complete package itself, the service, the pods, so there's nothing to port-forward to

Reply:
I would still recommend killing the process intentionally and immediately once we don't need it, to ease the development flow and UX. But in CI/CD I think it should be mostly fine, yeah, as CI/CD environment cleans up while it's completely deleted